From 56b624f6e8c844c954b84dd6ae58d455b48a5954 Mon Sep 17 00:00:00 2001
From: "Gupta Kumar, Sanjiv" <sanjiv.kumar.gupta@intel.com>
Date: Thu, 10 Aug 2017 16:14:29 +0530
Subject: [PATCH 4/4] Changes to make config environment variables available
 while parsing blueprints files

Change-Id: I6dc9017e5f26ed67e1536a583375bd22b84c588b
Category: Device Enablement
Origin: Internal
Upstream-Candidate: no, proprietary
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-48718
Signed-off-by: Gupta Kumar, Sanjiv <sanjiv.kumar.gupta@intel.com>
---
 core/main.mk | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/core/main.mk b/core/main.mk
index 3d42c8f..9b57355 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -9,6 +9,10 @@ else
 SHELL := /bin/bash
 endif
 
+TOP_DIR :=
+BUILD_SYSTEM := $(TOPDIR)build/core
+
+
 ifndef KATI
 
 host_prebuilts := linux-x86
@@ -24,6 +28,15 @@ run_soong_ui:
 $(sort $(MAKECMDGOALS)) : run_soong_ui
 	@#empty
 
+#Include config.mk even if KATI is not-defined, so that we set up standard variables based on configuration
+#We have enabled blueprint framework to understand environment variables
+#So including it here will make the environment variables available when bp files are being parsed
+include $(BUILD_SYSTEM)/config.mk
+ifdef VENDOR_ART_PATH
+export ART_TO_VENDOR_ART_PATH := ../$(VENDOR_ART_PATH)
+endif
+
+
 else # KATI
 
 # Absolute path of the present working direcotry.
@@ -32,9 +45,6 @@ else # KATI
 PWD := $(shell pwd)
 
 TOP := .
-TOPDIR :=
-
-BUILD_SYSTEM := $(TOPDIR)build/core
 
 # This is the default target.  It must be the first declared target.
 .PHONY: droid
@@ -48,6 +58,7 @@ droid_targets:
 # and host information.
 include $(BUILD_SYSTEM)/config.mk
 
+
 ifneq ($(filter $(dont_bother_goals), $(MAKECMDGOALS)),)
 dont_bother := true
 endif
-- 
1.9.1

